feat(plugin-calendar): Decrypt mercury calendar events

urbenlegend opened the pull request:
PR to get mercury calendar events properly decrypting in SDK.

Still need to write a mercury test for this but wanted to submit this PR for review first

@saurjai3 @ianwremmel

Commit:
chore(tooling): stop using grunt to build packages

Commit:
chore(tooling): remove coverage helper files

Commit:
chore(tooling): point test files at package main instead of helper

Commit:
chore(tooling): implement first pass at test.js

Commit:
chore(tooling): use yargs to limit browser or node; parallelize

ianwremmel committed:
chore(tooling): use argv to specify unit, int, auto

ianwremmel committed:
chore(tooling): use a single server for all automation tests

ianwremmel committed:
chore(tooling): use a single server for all automation tests

ianwremmel committed:
chore(tooling): make defaults readable by dotenv

ianwremmel committed:
chore(tooling): add --grep to npm test

ianwremmel committed:
chore(tooling): make watch mode available for karma

ianwremmel committed:
chore(tooling): test all packages; add --package

ianwremmel committed:
chore(tooling): launch test server automatically

ianwremmel committed:
chore(tooling): exit with code zero when xunit set

ianwremmel committed:
docs(tooling): document how to build and test with npm scripts

ianwremmel committed:
chore(tooling): remove unused files

ianwremmel committed:
chore(tooling): add debug statements and set mocha timeout

Commit:
refactor(tooling): avoid testsuite crosstalk

Commit:
refactor(all): unref timers to avoid stuck processes

Fixes #352

Commit:
test(all): deal with moved package locations

Commit:
refactor(tooling): point docker file at new test command

Commit:
chore(tooling): remove grunt:clean reference

Commit:
chore(tooling): put back grunt script for now

Commit:
docs(tooling): add jsdoc

Commit:
Merge branch 'master' into npm-test

* master:
  docs(general): update build commands
  docs(general): update Markdown syntax and build command
  docs(general): update README and CONTRIBUTING
  v0.7.100
  fix(plugin-calendar): Make test work in different env
  v0.7.99
  refactor(@ciscospark/plugin-phone): do not call /media twice
  refactor(@ciscospark/plugin-phone): always use modified offer sdp
  refactor(@ciscospark/plugin-phone): reduce locus polling
  style(@ciscospark/plugin-phone): alphabetize call.js

Commit:
chore(tooling): adjust dep check entry points

Commit:
refactor(tooling): use bash to check for karma xml

Commit:
fix(tooling): set coverage var safely

Commit:
fix(tooling): do coverage differently

Commit:
Merge branch 'master' into npm-test

* master:
  docs(general): update build commands
  docs(general): update Markdown syntax and build command
  docs(general): update README and CONTRIBUTING
  v0.7.100
  fix(plugin-calendar): Make test work in different env
  v0.7.99
  refactor(@ciscospark/plugin-phone): do not call /media twice
  refactor(@ciscospark/plugin-phone): always use modified offer sdp
  refactor(@ciscospark/plugin-phone): reduce locus polling
  style(@ciscospark/plugin-phone): alphabetize call.js

Commit:
docs(tooling): add missing jsdoc

Commit:
Merge branch 'npm-test' of github.com:ianwremmel/spark-js-sdk into npm-test

* 'npm-test' of github.com:ianwremmel/spark-js-sdk:
  docs(tooling): add missing jsdoc
  fix(tooling): do coverage differently
  fix(tooling): set coverage var safely

Commit:
fix(tooling): fix missing whitespace

Commit:
docs(@ciscospark/spark-core): clean up docs; prepare readme

Commit:
feat(@ciscospark/spark-core): make Token a core component

Commit:
test(@cisccospark/common): disable test in browser due to ie bug

ianwremmel committed:
fix(@ciscospark/test-helper-mock-spark): do not store instance

ianwremmel committed:
feat(@ciscospark/test-helper-mock-spark): add optional console output

ianwremmel committed:
fix(@ciscospark/spark-core): stop writing empty values to storage

ianwremmel committed:
feat(@ciscospark/spark-core): add Credentials as core component

ianwremmel committed:
test(@ciscospark/spark-core): add initial token tests

Commit:
fix(@@ciscospark/bin-sauce-connect): correct .sauce path

Commit:
chore(tooling): inject babel-polyfill into tests

Commit:
chore(tooling): unref all potential timers when the test suite closes

Commit:
feat(tooling): split out automation tests due to code coverage src

Commit:
chore(tooling): add package-specific fixes for tooling

Commit:
feat(tooling): force NODE_ENV to test

Commit:
chore(tooling): load env differently

Commit:
feat(tooling): use a requirehook to load covered files

Commit:
test(@ciscospark/plugin-credentials): point fixture at correct path

Commit:
test(@ciscospark/plugin-phone): handle new url scheme

Commit:
fix(tooling): use correct names for suite, report checker

Commit:
fix(tooling): use env vars that yargs will understand

ianwremmel committed:
test(@ciscospark/plugin-credentials): include babel-polyfill in tests

ianwremmel committed:
refactor(@ciscospark/plugin-wdm): eliminate auth interceptor

ianwremmel committed:
refactor(ciscospark): eliminate auth interceptor

ianwremmel committed:
feat(@ciscospark/spark-core): use supertoken when adequate

ianwremmel committed:
chore(deps): update sinon, chai

ianwremmel committed:
refactor(@ciscospark/spark-core): eliminate unneeded tests

ianwremmel committed:
feat(@ciscospark/spark-core): consolidate auth interceptor logic

ianwremmel committed:
feat(@ciscospark/spark-core): accept many shapes of "access token"

Commit:
fix(tooling): produce cobertura report

Commit:
fix(tooling): do not include legacy in code coverage

Commit:
refactor(all): use inBrowser instead of a window check

Commit:
feat(@ciscospark/common): add a constant "inBrowser"

Commit:
fix(tooling): do not include legacy in code coverage

Commit:
fix(tooling): alter pattern for junit files from which to strip stdout

Commit:
refactor(@ciscospark/spark-core): use authorizeUrl instead

Commit:
refactor(@ciscospark/spark-core): implement integration tests

Commit:
test(@ciscospark/spark-core): eliminate automation tests

Automation tests will be reimplemented in the browser credentials
plugins

Commit:
fix(@ciscospark/spark-core): address bug in serialize(); update tests

Commit:
feat(@ciscospark/spark-core): add #inspect() to SparkCore

Commit:
feat(@ciscospark/spark-core): make MemoryStoreAdapter preloadable

Commit:
test(@ciscospark/spark-core): confirm "loaded" event still fires

Commit:
refactor(@ciscospark/spark-core): implement "loaded" as an ampstate prop

Commit:
feat(@ciscospark/spark-core): implement "ready" event

Commit:
docs(@ciscospark/spark-core): document loaded, ready events

Commit:
feat(plugin-calendar): Decrypt mercury calendar events

Decrypt calendar mercury events

Commit:
feat(plugin-calendar): Decrypt mercury calendar events

Decrypt calendar mercury events

Commit:
fix(@ciscospark/spark-core): ensure ready depends on all children

Commit:
feat(@ciscospark/spark-core): add an onBeforeLogout hook

saurjai3 committed:
feat(spark-core): triggers request log events to the client

ianwremmel commented on packages/node_modules/@ciscospark/plugin-calendar/src/index.js:
```diff
@@ -36,6 +36,16 @@ registerPlugin(`calendar`, Calendar, {
         }
+      },
+      {
+        name: `transformMeetingFromMercury`,
```
This should be achievable without a new transformation. this predicate should be also be named `tranformMeeting`, but should provide a different `extract()` than the `transformMeeting` above.

urbenlegend replied:
Ah that makes sense. Done.

saurjai3 committed:
style(spark-core): logs the warnings on failure

saurjai3 committed:
Merge branch 'v7.2-consolidated' into v7.2-requestlogs

Commit:
feat(@ciscospark/plugin-authorization-node): add node oauth support

ianwremmel reviewed (changes requested):
Can this be tested?

urbenlegend commented:
Yes, I am writing a test now. Getting issues where I am not getting a mercury response back. Looking into it.

ianwremmel commented on packages/node_modules/@ciscospark/plugin-calendar/test/integration/spec/calendar.js:
```diff
@@ -78,9 +78,9 @@ function makeMockMeeting(params) {
+    let createdMeeting, creator, mccoy, meetingParams, mercuryPromise, spark, spock;
 
-    before(`create test users`, () => testUsers.create({
+    before(`create test users and initialize spark`, () => testUsers.create({
```
I would break this into multiple named before blocks. that makes it a lot easier to debug issues in jenkins.

ianwremmel commented on packages/node_modules/@ciscospark/plugin-calendar/test/integration/spec/calendar.js:
```diff
@@ -108,7 +108,19 @@ describe(`plugin-calendar`, () => {
+      // Setup mercury and callbacks
+      .then(() => spark.mercury.connect())
+      .then(() => {
+        mercuryPromise = new Promise((resolve) => {
```
This should probably go in the `it` blocks

ianwremmel commented on packages/node_modules/@ciscospark/plugin-calendar/test/integration/spec/calendar.js:
```diff
@@ -176,6 +186,25 @@ describe(`plugin-calendar`, () => {
+    );
+
+    it(`receives mercury event for created meeting`, () => {
+      return assert.isFulfilled(mercuryPromise)
```
How does the meeting get created?

saurjai3 committed:
refactor(spark-core): removed RequestEventInterceptor from post

urbenlegend commented:
Comments have been addressed @ianwremmel

ianwremmel committed:
refactor(@ciscospark/spark-core): export grantErrors

ianwremmel committed:
refactor(@ciscospark/spark-core): create grant errors from qs or res

ianwremmel committed:
feat(@ciscospark/spark-core): expose window via SparkCore (for mocking)

ianwremmel committed:
fix(@ciscospark/spark-core): remove clientType from core config

ianwremmel committed:
refactor(@ciscospark/plugin-authorization-node): use set()

ianwremmel committed:
feat(@ciscospark/spark-core): call auth.logout() if available

ianwremmel committed:
feat(@ciscospark/plugin-authorization-browser): implement first pass

ianwremmel committed:
test(@ciscospark/plugin-authorization-browser): test code and refresh

ianwremmel committed:
feat(@ciscospark/test-helper-server): add token refresh proxy

Commit:
feat(@ciscospark/plugin-authorization-browser): prevent parsing url

Commit:
feat(@ciscospark/plugin-wdm): unregister the device before logout

Commit:
feat(@ciscospark/plugin-mercury): disconnect from mercury  before logout

ianwremmel committed:
fix(@ciscospark/sparkcore): invalidate tokens correctly

ianwremmel committed:
style(@ciscospark/plugin-authorization-browser): tweak code sytling

ianwremmel committed:
feat(@ciscospark/test-helper-server): use state to prevent code exchange

ianwremmel committed:
feat(@ciscospark/spark-core): adjust error processing

ianwremmel committed:
feat(@ciscospark)/plugin-a-b-first-party): add browser code exchange

ianwremmel committed:
feat(@ciscospark/r-p-web-client): add auth plugin

ianwremmel committed:
feat(@ciscospark/spark-core): remove old credentials plugin

ianwremmel committed:
feat(@ciscospark/plugin-credentials): eliminate old plugin

ianwremmel committed:
feat(@ciscospark/plugin-authorization): add proxy for auth plugins

ianwremmel committed:
feat(ciscospark): add dependency on plaugin-authorization

ianwremmel committed:
feat(@ciscospark/plugin-machine-account): eliminate old plugin

ianwremmel committed:
fix(@ciscospark/spark-core): support additional data shape

ianwremmel committed:
feat(@ciscospark/plugin-encryption): update for credentials changes

ianwremmel committed:
feat(@ciscospark/plugin-board): update for credentials changes

ianwremmel committed:
feat(@ciscospark/test-helper-mock-spark): add getUserToken mock

ianwremmel committed:
feat(@ciscospark/spark-core): add getClientToken()

ianwremmel committed:
feat(@ciscospark/plugin-user): move hasPassword to user plugin

ianwremmel committed:
feat(@ciscospark/plugin-support): uset getClientToken

Commit:
feat(@ciscospark/plugin-encryption): add resiliency for kms auth failure

Commit:
feat(tooling): apply babel-register to src

Commit:
feat(payload-transform): attempt to decrypt error responses

Commit:
test(ciscospark): assume tests must instantiate CiscoSpark

Commit:
test(@ciscospark/spark-core): prove we can start without a token

Commit:
fix(@ciscospark/spark-core): do not downscope to same scope

Commit:
feat(ciscospark): remove automatic initialization

BREAKING CHANGE:  require('ciscospark') will no longer return a
CiscoSpark instance, but instead, the CiscoSpark constructor

Commit:
feat(ciscospark): add a oneliner for quickstarting shell scripts

ianwremmel commented on packages/node_modules/@ciscospark/plugin-calendar/test/integration/spec/calendar.js:
```diff
@@ -76,9 +76,31 @@ function makeMockMeeting(params) {
 describe(`plugin-calendar`, () => {
-  describe(`#list()`, () => {
-    let createdMeeting, creator, mccoy, meetingParams, spark, spock;
+  describe(`list meeting`, () => {
```
Why the name change here? describe blocks should typically adhere to the form:

```js
describe(`<plugin name>`, () => {
  describe(`<class name>`, () => {
    describe(`#<function name>()`, () => {
    });
  });
});
```

so, these tests should be under

```js
describe(`plugin-calendar`, () => {
  describe(`Calendar`, () => {
    describe(`#list()`, () => {
    });
  });
});
```

urbenlegend replied:
Okay, I'll change it back.

ianwremmel replied:
Note that it's actually always been missing a level.

urbenlegend replied:
Fixed

Commit:
style(plugin-calendar) Fix styling of describe blocks

Fix styling of describe blocks in calendar integration tests

Commit:
style(plugin-calendar) Fix styling of describe blocks

Fix styling of describe blocks in calendar integration tests

urbenlegend commented:
Thanks for the review @ianwremmel!

saurjai3 committed:
Merge branch 'calendar_mercury_decrypt' of github.com:urbenlegend/spark-js-sdk into v7.2-requestlogs

Commit:
feat(tooling): allow overriding browsers

Commit:
fix(@ciscospark/plugin-encryption): decrypt DRY errors

ianwremmel committed:
docs(all): update docs for auth/init changes

ianwremmel committed:
style(@ciscospark/spark-core): remove unneeded console statement

ianwremmel committed:
Merge branch 'npm-test' into merge-test

* npm-test: (54 commits)
  fix(@ciscospark/plugin-encryption): decrypt DRY errors
  style(plugin-calendar) Fix styling of describe blocks
  feat(plugin-calendar): Decrypt mercury calendar events
  feat(tooling): allow overriding browsers
  feat(payload-transform): attempt to decrypt error responses
  feat(tooling): apply babel-register to src
  feat(@ciscospark/plugin-encryption): add resiliency for kms auth failure
  fix(tooling): alter pattern for junit files from which to strip stdout
  fix(tooling): do not include legacy in code coverage
  fix(tooling): do not include legacy in code coverage
  fix(tooling): produce cobertura report
  test(@ciscospark/plugin-credentials): include babel-polyfill in tests
  fix(tooling): use env vars that yargs will understand
  fix(tooling): use correct names for suite, report checker
  test(@ciscospark/plugin-phone): handle new url scheme
  test(@ciscospark/plugin-credentials): point fixture at correct path
  feat(tooling): use a requirehook to load covered files
  chore(tooling): load env differently
  feat(tooling): force NODE_ENV to test
  chore(tooling): add package-specific fixes for tooling
  ...

ianwremmel committed:
refactor(@ciscospark/plugin-encryption): log better in test mode

ianwremmel committed:
fix(all): remove rogue "browser" hashes

ianwremmel committed:
feat(@ciscospark/t-h-refresh-callback): add default refreshCallback

Commit:
feat(tooling): log running docker containers each time one exits

Commit:
refactor(@ciscospark/spark-core): make plugin system more amp-friendly

Commit:
feat(all): add versions to core, ciscospark, and all plugins

Commit:
feat(tooling): do not run legacy tests in validated merge pipeline

Commit:
fix(all): reference plugins differently in tests

Commit:
fix(tooling): remove references to grunt:concurrent

Commit:
fix(tooling): move npm commands inside node container

Commit:
#release 0.8.0 #force-publish

Commit:
fix(tooling): fix breaking change detection

Commit:
chore(deps): install latest lerna

Commit:
#release 0.8.0 #force-publish

Commit:
fix(tooling): strip xunit logs when the build gets aborted

Commit:
feat(tooling): add checker to detect sauce tunnel failures

Commit:
chore(@ciscospark/plugin-phone): reduce timeout failure likelihood

Commit:
#release 0.8.0 #force-publish

Commit:
test(@ciscospark/plugin-calendar): retry whistler calls

Commit:
#release 0.8.0 #force-publish

Commit:
feat(tooling): add exec command

Commit:
feat(tooling): implement build via yargs

Commit:
feat(tooling): add command, lib helper for listing deps

Commit:
feat(tooling): implement dependent lookup

Commit:
feat(tooling): implement updated command

ianwremmel committed:
feat(tooling): implement version next and last

ianwremmel committed:
feat(tooling): implement version set

ianwremmel committed:
test(all): remove coverage helper

ianwremmel committed:
fix(tooling): make npm scripts windows friendlier

Fixes #577

ianwremmel committed:
feat(tooling): let version set rely on git to determine --all

ianwremmel committed:
feat(tooling): remove lerna

ianwremmel committed:
#release v0.8.0 #force-publish

Commit:
fix(tooling): change order of args and hope

Commit:
#release v0.8.0 #force-publish

Commit:
v0.8.0

ianwremmel committed:
style(tooling): clean lint

Commit:
vv0.8.0

Commit:
fix(tooling): detect change type with startsWith instead of regex

Commit:
#release v0.8.1 #force-publish

pcl committed:
test(@ciscospark/plugin-conversation): remove stickies tests

Commit:
v0.8.1

Commit:
fix(tooling): fix version detection/assignment

Commit:
Merge pull request #582 from pcl/peel-off-stickies

peel off stickies

* pcl/peel-off-stickies:
  test(@ciscospark/plugin-conversation): remove stickies tests

Commit:
chore(@ciscospark/plugin-conversation): add todo

Commit:
feat(@ciscospark/plugin-phone): use server-driven ringing signal

Commit:
v0.8.2

Commit:
feat(tooling): add "npm run clean"

Commit:
feat(@ciscospark/spark-core): throw if #when is used incorrectly

Commit:
test(@ciscospark/plugin-phone): use #when correctly

Commit:
feat(@ciscospark/plugin-phone): remove media stream urls

BREAKING CHANGE: this change will cause apps referencing
`call.remoteMediaStreamUrl` or `call.localMediaStreamUrl` to stop rendering
the referenced streams. Apps should rely on `call.removeMediaStream` and
`call.localMediaStream`

Commit:
fix(tooling): ensure we can compile all src on the fly

Commit:
fix(tooling): override browsers correctly

Commit:
v0.9.0

Commit:
fix(tooling): suppress publish errors

saurjai3 committed:
Merge branch 'master' into v8-consolidated

# Conflicts:
#	packages/node_modules/@ciscospark/plugin-avatar/src/avatar-url-store.js
#	packages/node_modules/@ciscospark/plugin-calendar/test/integration/spec/calendar.js
#	packages/node_modules/@ciscospark/plugin-credentials/src/credentials/common.js
#	packages/node_modules/@ciscospark/plugin-credentials/src/token.js
#	packages/node_modules/@ciscospark/plugin-credentials/test/integration/spec/token.js
#	packages/node_modules/@ciscospark/plugin-credentials/test/unit/spec/credentials.js
#	packages/node_modules/@ciscospark/spark-core/src/interceptors/auth.js
#	packages/node_modules/@ciscospark/spark-core/src/spark-core.js
#	packages/node_modules/@ciscospark/spark-core/test/unit/spec/interceptors/spark-tracking-id.js

Commit:
feat(@ciscospark/plugin-encryption): retry DRY errors in test mode

saurjai3 committed:
style(spark-core): linting

Commit:
fix(tooling): address syntax error

Commit:
#release v0.9.0

Commit:
v0.9.0

Commit:
fix(tooling): return updated dependents correctly

Commit:
feat(tooling): move test to new command structure

Commit:
#release v0.10.0 #force-publish

Commit:
v0.10.0

saurjai3 committed:
Merge branch 'master' into v8-consolidated

saurjai3 committed:
refactor(storage-adapter): removes web-client-internal on logout

saurjai3 committed:
style(storage-adapter): removes bounded storage key on logout

saurjai3 committed:
style(plugin-conversation): returns failure message on decryption

Commit:
Release v0.5.344

Commit:
v0.10.1
